<a href="https://colab.research.google.com/github/Jaejuna/SpeedWagon/blob/main/Data_extract(1_10).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd

In [3]:
train = pd.read_csv('/content/drive/MyDrive/BOAZ_miniProject2/train.csv')
valid = pd.read_csv('/content/drive/MyDrive/BOAZ_miniProject2/valid.csv')

In [4]:
print("Size of train data :", len(train))
print("Size of validation data :", len(valid))

Size of train data : 279992
Size of validation data : 35004


In [5]:
train['Topic'].unique()

array(['주거와생활', '시사교육', '상거래(쇼핑)', '개인및관계',
       '행사', '미용과건강', '일과직업', '식음료', '여가생활'],
      dtype=object)

In [6]:
# Topic 별 Dialogue 수
pd.DataFrame(train.groupby("Topic").count().iloc[:,0])

Text
Topic                
개인및관계   71130
미용과건강   17069
상거래(쇼핑)  26298
시사교육       13119
식음료        30184
여가생활      35247
일과직업     20428
주거와생활    45179
행사           21338

In [7]:
pd.DataFrame(valid.groupby("Topic").count().iloc[:,0])

Text
Topic               
개인및관계   8892
미용과건강   2134
상거래(쇼핑)  3288
시사교육       1640
식음료        3773
여가생활      4407
일과직업     2554
주거와생활    5648
행사           2668

# Data extract
- 토픽 9개
- train : 279992 -> 27993(1/10로 줄임)
  - 개인및관계(7113), 미용과건강(1706), 상거래(2629), 시사교육(1311), 식음료(3018), 여가생활(3524), 일과직업(2042), 주거와생활(4517), 행사(2133)
- valid : 35004 -> 3496(1/10로 줄임)
  - 개인및관계(889), 미용과건강(213), 상거래(328), 시사교육(164), 식음료(377), 
  여가생활(440), 일과직업(255), 주거와생활(564), 행사(266)  
- 랜덤으로 추출
 

## train data

In [33]:
df1 = train.loc[train['Topic']=='주거와생활'].sample(n=4517)
df2 = train.loc[train['Topic']=='시사교육'].sample(n=1311)
df3 = train.loc[train['Topic']=='상거래(쇼핑)'].sample(n=2629)
df4 = train.loc[train['Topic']=='개인및관계'].sample(n=7113)
df5 = train.loc[train['Topic']=='행사'].sample(n=2133)
df6 = train.loc[train['Topic']=='미용과건강'].sample(n=1706)
df7= train.loc[train['Topic']=='일과직업'].sample(n=2042)
df8 = train.loc[train['Topic']=='식음료'].sample(n=3018)
df9 = train.loc[train['Topic']=='여가생활'].sample(n=3524)

In [47]:
 train_new = pd.concat([df1,df2,df3,df4,df5,df6,df7,df8,df9], ignore_index=True)

## valid data

In [48]:
df1 = valid.loc[valid['Topic']=='주거와생활'].sample(n=564)
df2 = valid.loc[valid['Topic']=='시사교육'].sample(n=164)
df3 = valid.loc[valid['Topic']=='상거래(쇼핑)'].sample(n=328)
df4 = valid.loc[valid['Topic']=='개인및관계'].sample(n=889)
df5 = valid.loc[valid['Topic']=='행사'].sample(n=266)
df6 = valid.loc[valid['Topic']=='미용과건강'].sample(n=213)
df7= valid.loc[valid['Topic']=='일과직업'].sample(n=255)
df8 = valid.loc[valid['Topic']=='식음료'].sample(n=377)
df9 = valid.loc[valid['Topic']=='여가생활'].sample(n=440)

In [49]:
valid_new = pd.concat([df1,df2,df3,df4,df5,df6,df7,df8,df9], ignore_index=True)

## save

In [53]:
train_new.to_csv('/content/drive/MyDrive/BOAZ_miniProject2/train_new.csv', index = False)

In [52]:
valid_new.to_csv('/content/drive/MyDrive/BOAZ_miniProject2/valid_new.csv', index = False)